ssl/statem/extensions_clnt.c

```diff
@@ -31,6 +31,28 @@ int tls_construct_ctos_renegotiate(SSL *s, WPACKET *pkt, unsigned int context,
     return 1;
 }
 
+int tls_construct_ctos_maxfragmentlen(SSL *s, WPACKET *pkt, unsigned int context,
+                                      X509 *x, size_t chainidx, int *al)
+{
+    /* Add Max Fragment Length extension if user enabled it. */
+    if (s->ext.max_fragment_length != TLSEXT_max_fragment_length_DISABLED) {
```

Comment:
Yes it is set in the SSL_SESSION structure, so this extension will stay active after session resumption.